core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -230,12 +251,16 @@ class NewHadoopRDD[K, V](
     super.persist(storageLevel)
   }
 
-
-  def getConf: Configuration = confBroadcast.value.value
```

Comment:
My concern wasn't perf. impact so much as whether people might be calling `getConf` and expecting to be able to mutate a configuration instance that was also used somewhere else.